include/zephyr/bluetooth/addr.h

```diff
@@ -81,6 +81,14 @@ static inline int bt_addr_cmp(const bt_addr_t *a, const bt_addr_t *b)
 	return memcmp(a, b, sizeof(*a));
 }
 
+/** @brief Compare Bluetooth LE device addresses for equality.
+ */
+static inline bool bt_addr_le_eq(const bt_addr_le_t *a, const bt_addr_le_t *b)
+{
+	return ((a->type == b->type)
+	        && (memcmp(a->a.val, b->a.val, sizeof(a->a.val)) == 0));
```

Comment:
Same thing here with the placement of `&&`

Reply:
This will disappear when I rebase on https://github.com/zephyrproject-rtos/zephyr/pull/50361.